setup.sh

```diff
@@ -122,101 +156,131 @@ if [[ -z "$SKIP_SEEDING" ]]; then
         --config=./content-api/data/cloudbuild.yaml
     fi
 fi # skip seeding
 
-####################
-# Build Containers #
-####################
-
-SETUP_IMAGE_TAG="setup"
-E2E_RUNNER_TAG="latest"
-
-if [[ -z "$SKIP_BUILD" ]]; then
-
-echo
-echo "$(tput bold)Building container images for testing and application hosting...$(tput sgr0)"
-echo
-
-gcloud builds submit "content-api" \
-    --config=ops/api-build.cloudbuild.yaml \
-    --project="$OPS_PROJECT" --substitutions=_REGION="$REGION",_IMAGE_TAG="$SETUP_IMAGE_TAG",_CONTEXT="."
-
-gcloud builds submit \
-    --config=ops/web-build.cloudbuild.yaml \
-    --ignore-file=ops/web-build.gcloudignore \
-    --project="$OPS_PROJECT" --substitutions=_REGION="$REGION",_IMAGE_TAG="$SETUP_IMAGE_TAG"
-
-gcloud builds submit "ops/e2e-runner" \
-    --config=ops/e2e-runner-build.cloudbuild.yaml \
-    --project="$OPS_PROJECT" --substitutions=_REGION="$REGION",_IMAGE_TAG="$E2E_RUNNER_TAG"
-
-fi # skip build
-
-
 ##################
 # Deploy Services #
 ##################
 
+# Function will check build before deploying Run service
+check_for_build_then_run () {
+    local build_id="${1}"
+    local run_command="${2}"
+    # Wait for build to complete.
+    if [ $(gcloud builds describe $build_id --format='value(status)') == "WORKING" ]; then
+        log_url=$(gcloud builds describe $build_id --format='value(logUrl)')
+        echo "Build $build_id still working."
+        echo "Logs are available at [ $log_url ]."
+        while [ $(gcloud builds describe $build_id --format='value(status)') == "WORKING" ]
+        do
+          echo "Build $build_id still working..."
+          sleep 10
+        done
+    fi
+    # Return error and build log for failures. 
+    if [ $(gcloud builds describe $build_id --format='value(status)') == "FAILURE" ]; then
+        build_describe=$(gcloud builds describe $build_id --format='csv(failureInfo.detail,logUrl)[no-heading]')
+        fail_info=$(echo $build_describe | awk -F',' '{gsub(/""/,"\"");print $1}')
+        log_url=$(echo $build_describe | awk -F',' '{print $2}')
+        echo "Build ${build_id} failed. See build log: $log_url"
+        echo "ERROR: ${fail_info}"
+        echo "Please re-run setup."
+        exit 2
+    # Deploy if build is successful.
+    elif [ `gcloud builds describe $build_id --format='value(status)'` == "SUCCESS" ]; then
+        $run_command
+    # Return build log for all other statuses.
+    else
+        log_url=$(gcloud builds describe $build_id --format='value(logUrl)')
+        echo "Build ${build_id} did not complete." 
+        echo "See build log: $log_url"
+        echo "Please re-run setup."
+        exit 2
+    fi;
+}
+
 if [[ -z "$SKIP_DEPLOY" ]]; then
 
     echo
     echo "$(tput bold)Deploying Cloud Run services...$(tput sgr0)"
     echo
 
     ## Staging Services ##
-
-    gcloud run deploy content-api \
+    check_for_build_then_run $API_BUILD_ID "gcloud run deploy content-api \
```

Comment:
I'm a bit concerned that converting swathes of our operations into function arguments is going to make this less maintainable.

Instead, could we add the delay as a self-contained step before running the gcloud command? That way we keep the flow control more flat. I'm not seeing that you're relying on much from passthru.

Reply:
I made it into a function to avoid having to repeat the same lines of code for each run deployment.  I've added extra comments to the function definition to hopefully make this more maintainable in the future